EduNLP/Pretrain/quesNet_vec.py

```diff
@@ -0,0 +1,781 @@
+        if isinstance(item, Exception):
+            raise item
+        else:
+            self.pos += 1
+            return item
+
+    def produce(self):
+        for i in range(self.pos, len(self.index)):
+            try:
+                index = self.index[i]
+
+                bs = self.batch_size
+
+                if callable(self.data):
+                    data_batch = self.data(index * bs, (index + 1) * bs)
+                else:
+                    data_batch = self.data[index * bs:(index + 1) * bs]
+
+                label_batch = [label[index * bs:(index + 1) * bs]
+                               for label in self.label]
+                if label_batch:
+                    self.queue.put([data_batch] + label_batch)
+                else:
+                    self.queue.put(data_batch)
+            except Exception as e:
+                self.queue.put(e)
+                return
+
+
+class EmbeddingDataset(Dataset):
+    def __init__(self, data, data_type='image'):
+        self.data = data
+        self.data_type = data_type
+        assert self.data_type in ['image', 'meta']
+
+    def __len__(self):
+        return len(self.data)
+
+    def __getitem__(self, idx):
+        if self.data_type == 'image':
+            return to_tensor(self.data[idx])
+        elif self.data_type == 'meta':
+            return torch.tensor(self.data[idx])
+
+
+def pretrain_iter(ques, batch_size):
+    _cur_iter = PrefetchIter(ques, batch_size=batch_size)
+    return _cur_iter
+
+
+sigint_handler = signal.getsignal(signal.SIGINT)
+
+
+def critical(f):
+    it = iter(f)
+    signal_received = ()
+
+    def handler(sig, frame):
+        nonlocal signal_received
+        signal_received = (sig, frame)
+        logging.debug('SIGINT received. Delaying KeyboardInterrupt.')
+
+    while True:
+        try:
+            signal.signal(signal.SIGINT, handler)
+            yield next(it)
+            signal.signal(signal.SIGINT, sigint_handler)
+            if signal_received:
+                sigint_handler(*signal_received)
+        except StopIteration:
+            break
+
+
+def pretrain_embedding_layer(dataset: EmbeddingDataset, ae: AE, lr: float = 1e-3, log_step: int = 1, epochs: int = 3,
+                             batch_size: int = 4, device=torch.device('cpu')):
+    train_dataloader = DataLoader(dataset, batch_size=batch_size, shuffle=True)
+    optim = torch.optim.Adam(ae.parameters(), lr=lr)
+    ae.train()
+    ae.to(device)
+    train_type = dataset.data_type
+    for i in range(epochs):
+        for batch, item in tqdm(enumerate(train_dataloader)):
+            loss = ae.loss(item.to(device))
+            optim.zero_grad()
+            loss.backward()
+            optim.step()
+            if batch % log_step == 0:
+                logger.info(f"[Epoch{i}][Batch{batch}]Training {train_type} Embedding layer, loss:{loss}")
+    return ae
+
+
+def pretrain_QuesNet(path, output_dir, tokenizer, save_embs=False, train_params=None):
```

Comment:
use snake case or camel case naming style :
- Method name of  `pretrain_QuesNet` maybe `pretrain_quesnet`
- Module name `quesNet_vec`  maybe   `quesnet_vec`